core/tx_pool_test.go

```diff
@@ -326,6 +326,7 @@ func TestInvalidTransactions(t *testing.T) {
 
 	pool, key := setupTxPool()
 	defer pool.Stop()
+	pool.gfork = false
```

Comment:
Why shouldn't we add the case here of a gatewayFee bigger than zero?
If this test aims to check the whole validity of the transactions in the last fork, we should add the case too.
If we want to maintain this as testing just before the Gfork, I would change the name of the test, and maybe copy-paste it and change the copy to mimic a valid tx in the gFork.

My point here is that we should have a test that "checks every check" working in the last fork. Otherwise we can break one of the other checks in the gFork, without noticing (it's not the case, but I think that you get my point)